kc/test_dependencies/src/lib.rs

```diff
@@ -1,39 +1,91 @@
+use std::ffi::c_char;
+
+// FIXME: it shoudn't be necessary to duplicate the definition of the
+// valgrind_do_client_request_expr macro, but I had trouble dealing with tryign
+// to make a single definition that could handle the various ways one might need
+// to reference the VgKrabcakeClientRequestData struct. Look into the standard
+// ways to address this (perhaps Macros 2.0 or even 1.5 is the answer)
```

Comment:
I think we could move the nested macro out but we'd have to import it in each test that uses the outer macro. I also don't know why it behaves like that. But it might be preferable to duplicating the code (it doesn't matter for this review though, we can address it later)

Reply:
I figure when we rework this, maybe we'll try to switch to Macros 2.0, and for the meantime we'll live with duplicating the code here. (I'm pretty sure we're better off with this local bit of duplication rather than having to complicate every test.)

The other thing to consider is that this macro is all short term anyway. In the *non*-short term we need to move to having rustc inject these asm sequences. So most any investment we put into this is likely to be throw away sometime in the next month or so.